setup/install-pre-req.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+#This script sets up the pre-requisites for setting up app-studio in local machine.
+# mainly setting up pip3,yq, kubectl, kustomize
+# Firsly it checks which type of system is running this
+# Then it installs the dependencies one by one
+
+OSNAME=$(grep '^NAME=' /etc/os-release)
+echo $OSNAME
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..
+echo $ROOT
+
```

Comment:
Since it's installing crc there should be a check that there is enough RAM to proceed.

Reply:
That's a good point. We need to check for the RAM as we are setting the default MEMORY config to 16GB for running app-studio

Reply:
CodeReady Containers requires the following system resources:

4 physical CPU cores
9 GB of free memory
35 GB of storage space

But we'll definitely need more than that.  The question is how much.  I know that some of our developers are happy with these values and others (with beefier machines) set them higher.

Reply:
I think we can rely on the 'crc setup' and 'crc start' commands to do the checks.

Reply:
Moved to https://github.com/redhat-appstudio/infra-deployments/issues/214